File: server/src/main/java/org/dan/jadalnia/app/ws/WsHandlerConfigurator.java
package org.dan.jadalnia.app.ws;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.context.support.AnnotationConfigWebApplicationContext;

import javax.websocket.server.ServerEndpointConfig;

@Slf4j
public class WsHandlerConfigurator extends ServerEndpointConfig.Configurator {
    private static volatile AnnotationConfigWebApplicationContext injector;

    public static void setInjector(AnnotationConfigWebApplicationContext injector) {
        log.info("Set web socket injector {}", injector);
        WsHandlerConfigurator.injector = injector;
    }

    @Override
    public <T> T getEndpointInstance(Class<T> endpointClass) throws InstantiationException {
        try {
            return injector.getAutowireCapableBeanFactory().createBean(endpointClass);
        } catch (Exception e) {
            log.error("Failed to instantiate web socket handler {}", endpointClass, e);
            throw new InstantiationException("Failed to instantiate a web socket handler");
        }
    }

}

File: service/src/main/kotlin/org/dan/jadalnia/app/ws/WsHandlerConfigurator.kt
package org.dan.jadalnia.app.ws

import org.slf4j.LoggerFactory
import org.springframework.web.context.support.AnnotationConfigWebApplicationContext

import javax.websocket.server.ServerEndpointConfig

class WsHandlerConfigurator
    : ServerEndpointConfig.Configurator() {

    companion object {
        var _injector: AnnotationConfigWebApplicationContext? = null
        val log = LoggerFactory.getLogger(WsHandlerConfigurator::class.java)

        @JvmStatic
        fun setInjector(injector: AnnotationConfigWebApplicationContext) {
            log.info("Set web socket injector {}", injector)
            _injector = injector
        }

        fun getInjector() = _injector as AnnotationConfigWebApplicationContext
    }

    override fun <T> getEndpointInstance(endpointClass: Class<T>): T {
        try {
            return getInjector()
                    .getAutowireCapableBeanFactory()
                    .createBean(endpointClass)
        } catch (e: Exception) {
            log.error("Failed to instantiate web socket handler {}", endpointClass, e)
            throw InstantiationException("Failed to instantiate a web socket handler")
        }
    }
}
